common/test/src/ui/tst_MapDocument.cpp

```diff
@@ -462,6 +463,73 @@ TEST_CASE_METHOD(MapDocumentTest, "canUpdateLinkedGroups")
     document->canUpdateLinkedGroups(kdl::vec_static_cast<mdl::Node*>(entityNodes)));
 }
 
+TEST_CASE_METHOD(MapDocumentTest, "newDocument")
+{
+  const auto fgdContent = R"(
+    @SolidClass = worldspawn : "World entity" [
+      some_default_prop(string) : "" : "value"
+    ]
+  )";
+
+  const auto configTemplate = R"(
+    {{
+        version: 9,
+        name: "testNewDocument",
+        icon: "Icon.png",
+        "fileformats": [ {{ "format": "Standard" }} ],
+        "filesystem": {{
+            "searchpath": ".",
+            "packageformat": {{ "extension": ".pak", "format": "idpak" }}
+        }},
+        "materials": {{
+            "root": "textures",
+            "format": {{ "extensions": [".jpg"], "format": "image" }}
+        }},
+        "entities": {{
+            "definitions": [ "WorldSpawnOnly.fgd" ],
+            "defaultcolor": "0.6 0.6 0.6 1.0",
+            "setDefaultProperties": {}
+        }},
+        "tags": {{ "brush": [], "brushface": [] }},
+        "faceattribs": {{ "surfaceflags": [], "contentflags": [] }}
+    }}
+  )";
+
+  SECTION("Default worldspawn properties")
+  {
+    using T = std::tuple<bool, std::vector<mdl::EntityProperty>>;
+    const auto& [setDefaultProperties, expectedEntityProperties] = GENERATE(values<T>({
+      {false,
+       {{mdl::EntityPropertyKeys::Classname,
+         mdl::EntityPropertyValues::WorldspawnClassname}}},
+      {true,
+       {{mdl::EntityPropertyKeys::Classname,
+         mdl::EntityPropertyValues::WorldspawnClassname},
+        {"some_default_prop", "value"}}},
+    }));
+
+    const auto configContent =
+      fmt::format(fmt::runtime(configTemplate), setDefaultProperties ? "true" : "false");
```

Comment:
What compiler error do you get if you use `fmt::format`? I haven't had the need for `fmt::runtime` so far.

Reply:
Here a code where you can test it
https://godbolt.org/z/ve7d5aEha

Reply:
I see. If you declare `configTemplate` as `constexpr` it works and you can remove `fmt::runtime`: https://godbolt.org/z/oWTTdv5zh